executor/join.go

```diff
@@ -218,13 +221,42 @@ func (e *HashJoinExec) fetchOuterChunks(ctx context.Context) {
 			}
 			return
 		}
+
+		if !hasWaitedForInner {
```

Comment:
Why not do this before the for loop?

Reply:
actually we should do this after the first function call of `e.outerExec.Next()`

Reply:
ok, got it.